sageresearch-app-sdk/src/main/java/org/sagebionetworks/research/sageresearch_app_sdk/TaskResultUploader.java

```diff
@@ -61,22 +68,32 @@
     private AbstractResultArchiveFactory abstractResultArchiveFactory;
     private AccountDAO accountDAO;
     private UploadManager uploadManager;
+    private ScheduleRepository scheduleRepo;
 
     @Inject
     public TaskResultUploader(final BridgeConfig bridgeConfig,
                               final AbstractResultArchiveFactory abstractResultArchiveFactory,
                               final UploadManager uploadManager,
-                              final AccountDAO accountDAO) {
+                              final AccountDAO accountDAO,
+                              Context context) {
         this.bridgeConfig = bridgeConfig;
         this.abstractResultArchiveFactory = abstractResultArchiveFactory;
         this.uploadManager = uploadManager;
         this.accountDAO = accountDAO;
+        this.scheduleRepo = ScheduleRepository.Companion.getInstance(context);
     }
 
     @Override
     public Completable processTaskResult(final TaskResult taskResult) {
         LOGGER.info("Processing task result: {}", taskResult);
 
+        // TODO: mdephillips 10/2/18 convert to a completable and integrate into this flow to create metadata
+        scheduleRepo.findSchedule(taskResult.getTaskUUID(), scheduledActivityEntity -> {

```

Comment:
What goes here? Is this not something that should go in `ScheduledActivityTaskResultProcessor`?

Reply:
All data model interactions should go through ScheduleRepository.  Find schedule just looks up the schedule in the db.